myapp/app/controllers/tasks_controller.rb

```diff
@@ -1,6 +1,7 @@
 class TasksController < ApplicationController
   def index
     @tasks = Task.all
+    @tasks = Task.search_by_name_and_status(params[:name], params[:status]) if params[:name].present? || params[:status].present?
```

Comment:
[IMO]
This way of writing would require re-coding if the specification is changed to search for a different item.

I mean I want to write in controller just
```
@tasks = Task.dynamic_search(params || {})
```
and in model, maybe you can write scope like this
```
  scope :dynamic_search, ->(search_params) do
    query = all
    search_params.each do |key, value|
        ...
    end
    query
  end
```

Reply:
Thanks for your advice, After change to `dynamic_search` it would be more flexible, I have changed it.